core/src/main/scala/kafka/server/ForwardingManager.scala

```diff
@@ -109,7 +113,14 @@ class ForwardingManagerImpl(
         } else {
           parseResponse(envelopeResponse.responseData, requestBody, request.header)
         }
-        responseCallback(response)
+
+        // Unsupported version indicates an incompatibility between controller and client API versions. The
+        // forwarding broker should close the connection with the client and let it reinitialize the connection
+        // and refresh the controller API versions.
+        if (response.errorCounts().containsKey(Errors.UNSUPPORTED_VERSION)) {
```

Comment:
Taking into account #9850, I think we should be checking `envelopeError`. If the broker could not parse the embedded request, then it would not be able to include an embedded response.